### DISCUSSION TITLE
Support a way to align Blocks at the bottom of their container

### DISCUSSION THREAD

**Discussion by MadtownLems** [WordPress/gutenberg#48659] [General]
It's a pretty common/popular design concept to have some information in columns, with a call to action (or similar element) at the bottom. Ideally, these calls to action would be aligned with each other.   

![2023-03-01 15_16_44-Column Test – Topic Hub](https://user-images.githubusercontent.com/945157/222266900-85360dd5-0250-4e89-8277-82ad136b4f34.png)

Notice how the third "Learn More" button is much higher than the others.

Currently, I don't believe there is a core-supported way to force the "Learn More" buttons to be aligned in the bottom of these columns (without manual custom CSS). I'm trying to come up with how this would be implemented, or if there are any ongoing conversations around it that I can participate in.

Any ideas?

**Comment by skorasaurus** [comment]
I believe this type of design is only possible with CSS Grid.... Even with grid, i[t's a little tricky to do](https://ishadeed.com/article/learn-css-subgrid/) since the subgrid property is only implemented by firefox. 

There's discussion at https://github.com/WordPress/gutenberg/issues/47809 and you can cite your layout as an example.

**Reply by skorasaurus** [reply]
Remembered that this could be done with `align-items: end` on a parent block :)

**Comment by talldan** [comment]
How about this kind of structure:
- Stack (with min. height, central justification, space between alignment and padding set)
   - Stack (items aligned top)
       - Image
       - Title
       - Paragraph
   - Buttons
       - Button
       
 Here's a screenshot of how it looks in the editor:
![Screen Shot 2023-03-07 at 12 32 29 pm](https://user-images.githubusercontent.com/677833/223321334-7883e742-0a05-4dfd-a184-32b7cf584dbc.png)

       
@tellthemachines may also have some other suggestions.

**Reply by tellthemachines** [reply]
I'm thinking all the cards can be Stacks within a Row, and then we play with their justification, alignment, Width and Height controls until we get the exact layout:

<img width="678" alt="Screenshot 2023-03-07 at 4 09 33 pm" src="https://user-images.githubusercontent.com/8096000/223326568-2a596a08-5367-4eb2-b293-2a93e4d61142.png">

The code view for this would be:
```
<!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap","verticalAlignment":"stretch","justifyContent":"left"}} -->
<div class="wp-block-group"><!-- wp:group {"style":{"color":{"background":"#506475"},"spacing":{"padding":{"top":"var:preset|spacing|40","right":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40"}},"layout":{"selfStretch":"fixed","flexSize":"50%"}},"textColor":"contrast","layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
<div class="wp-block-group has-contrast-color has-text-color has-background" style="background-color:#506475;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:image {"id":7,"sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large"><img src="http://localhost:8888/wp-content/uploads/2023/02/charlie05-1024x768.jpg" alt="" class="wp-image-7"/></figure>
<!-- /wp:image -->

<!-- wp:heading -->
<h2 class="wp-block-heading">Card title</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>Cat ipsum dolor sit amet, meoooow. Lies down . Chase after silly colored fish toys around the house. I is playing on your console hooman walk on keyboard</p>
<!-- /wp:paragraph -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center","orientation":"horizontal"}} -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">a button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group -->

<!-- wp:group {"style":{"color":{"background":"#506475"},"spacing":{"padding":{"top":"var:preset|spacing|40","right":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40"}},"layout":{"selfStretch":"fixed","flexSize":"50%"}},"textColor":"contrast","layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
<div class="wp-block-group has-contrast-color has-text-color has-background" style="background-color:#506475;padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:image {"id":7,"sizeSlug":"large","linkDestination":"none"} -->
<figure class="wp-block-image size-large"><img src="http://localhost:8888/wp-content/uploads/2023/02/charlie05-1024x768.jpg" alt="" class="wp-image-7"/></figure>
<!-- /wp:image -->

<!-- wp:heading -->
<h2 class="wp-block-heading">Card title</h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"style":{"layout":{"selfStretch":"fill","flexSize":null}}} -->
<p>Cat ipsum dolor sit amet, meoooow. Lies down .</p>
<!-- /wp:paragraph -->

<!-- wp:buttons {"layout":{"type":"flex","justifyContent":"center","orientation":"horizontal"}} -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">a button</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:group -->
```

It's not a great experience currently because due to the flex implementation of Row, we have to set the width of each of its children explicitly to 50% 50%, or 33% 33% 33% etc. It would be nicer with grid, and definitely worth looking into as a future grid use case!

**Reply by skorasaurus** [reply]
Thanks for sharing the code and the example. 

This setup is a good start and suitable for some needs although I'm making a very similar layout to @MadtownLems and so far, Gutenberg's layout tools kind of fall apart when you have more than 3 'cards' (whether they're stacks or groups) in your example. 

For example, if you have 5 cards; they don't stretch across multiple lines and will appear squished on smaller viewpoints.
![image](https://user-images.githubusercontent.com/955351/223745897-b7dc275c-f722-40f1-97aa-da2af1a89234.png)

You can adjust the parent row container and add `flexWrap: wrap` as a block attribute and value (translating to the css rule `flex-wrap: wrap`) but you then only 1 card item will display horizontally even on larger viewpoints. 

![image](https://user-images.githubusercontent.com/955351/223747721-357a00eb-605b-4596-8e05-044d4b8aa874.png)

Ideally, where you could set the flex-basis on each card container [in this example that I made](https://codepen.io/skorasaurus/pen/yLEJQxE)

**Reply by tellthemachines** [reply]
You can set flex-basis on each card by using the "width" control under Dimensions in the sidebar:

<img width="278" alt="Screenshot 2023-03-09 at 3 16 14 pm" src="https://user-images.githubusercontent.com/8096000/223915495-c3343c0f-065d-41d3-85e0-7be4c704f1d1.png">

But I agree that the responsive behaviour is far from ideal. Flex is not very well suited to fine-grained responsiveness on containers with multiple children.

**Comment by DarkLite1** [comment]
Stumbled upon this same issue. Is there a native Gutenberg solution in the meantime? Like using "Group" and then within the group select "Aling to bottom" or something?

See the price not being aligned:

![image](https://github.com/user-attachments/assets/8e117477-c2c2-47e8-93a1-11c22459530b)

**Reply by hanneslsm** [reply]
Yes, create a stack and give it a 100% Min height and set height of the inner stack to "fill"
<img width="1515" alt="image" src="https://github.com/user-attachments/assets/8a129942-57fc-4ee6-abf9-eb8e62a64ee8">
<details><summary>Block Markup</summary>
<p>

```
<!-- wp:columns {"verticalAlignment":null} -->
<div class="wp-block-columns"><!-- wp:column {"verticalAlignment":"stretch","style":{"spacing":{"padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20","left":"var:preset|spacing|20","right":"var:preset|spacing|20"}}},"backgroundColor":"accent-4"} -->
<div class="wp-block-column is-vertically-aligned-stretch has-accent-4-background-color has-background" style="padding-top:var(--wp--preset--spacing--20);padding-right:var(--wp--preset--spacing--20);padding-bottom:var(--wp--preset--spacing--20);padding-left:var(--wp--preset--spacing--20)"><!-- wp:group {"style":{"dimensions":{"minHeight":"100%"}},"layout":{"type":"flex","orientation":"vertical"}} -->
<div class="wp-block-group" style="min-height:100%"><!-- wp:group {"style":{"layout":{"selfStretch":"fill","flexSize":null}},"layout":{"type":"flex","orientation":"vertical","verticalAlignment":"top"}} -->
<div class="wp-block-group"><!-- wp:heading -->
<h2 class="wp-block-heading">Heading</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>Lorem ipsum dolor sit amet. Aut repudiandae rerum </p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Click here</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:column -->

<!-- wp:column {"verticalAlignment":"stretch","style":{"spacing":{"padding":{"top":"var:preset|spacing|20","bottom":"var:preset|spacing|20","left":"var:preset|spacing|20","right":"var:preset|spacing|20"}}},"backgroundColor":"accent-4"} -->
<div class="wp-block-column is-vertically-aligned-stretch has-accent-4-background-color has-background" style="padding-top:var(--wp--preset--spacing--20);padding-right:var(--wp--preset--spacing--20);padding-bottom:var(--wp--preset--spacing--20);padding-left:var(--wp--preset--spacing--20)"><!-- wp:group {"style":{"dimensions":{"minHeight":"100%"}},"layout":{"type":"flex","orientation":"vertical","verticalAlignment":"space-between"}} -->
<div class="wp-block-group" style="min-height:100%"><!-- wp:group {"style":{"layout":{"selfStretch":"fill","flexSize":null}},"layout":{"type":"flex","orientation":"vertical","verticalAlignment":"top"}} -->
<div class="wp-block-group"><!-- wp:heading -->
<h2 class="wp-block-heading">Heading</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>Lorem ipsum dolor sit amet. Rem rerum tenetur est eligendi amet qui tenetur alias aut explicabo perspiciatis qui aspernatur consequatur. </p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button -->
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Click here</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:group --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->
```
</p>
</details>

**Reply by hanneslsm** [reply]
> Like using "Group" and then within the group select "Aling to bottom" or something?

Ah, I think I misread. This function as you have described it does not exist.

**Comment by DarkLite1** [comment]
Figured it out thanks to [this YouTube tutorial](https://youtu.be/bbMqs6gQxkc?si=NPuo6GUU1c45eaBF). 

In the Spectra plugin you have a "[Container block](https://wpspectra.com/docs/spectra-container-block/)" that has a lot more options than the "Columns block". These containers use flex classes and can easily be modified to align the content perfectly without any custom CSS or coding. 

For anyone struggling with this, please follow the video and you'll be fine.
